client/components/sub-masterbar-nav/index.jsx

```diff
@@ -0,0 +1,161 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { findDOMNode } from 'react-dom';
+import classNames from 'classnames';
+
+import Gridicon from 'gridicons';
+import Item from './item';
+
+const VIEW_BREAKPOINT = 660;
+const VIEW_PADDING = 50;
+const ITEM_WIDTH = 110;
+
+export default class SubMasterbarNav extends Component {
+	static propTypes = {
+		fallback: PropTypes.shape(
+			{
+				label: PropTypes.string.isRequired,
+				uri: PropTypes.string.isRequired,
+				icon: PropTypes.string
+			}
+		),
+		options: PropTypes.arrayOf(
+			PropTypes.shape( {
+				label: PropTypes.string.isRequired,
+				uri: PropTypes.string.isRequired,
+				icon: PropTypes.string
+			} )
```

Comment:
Might make sense to define this as an `OptionShape` `const` and re-use both here and for `fallback`.